File: src/main/java/com/futureclient/FutureClient.java
package com.futureclient;

import com.futureclient.config.ConfigManager;
import com.futureclient.module.ModuleManager;
import com.futureclient.gui.GuiManager;
import com.futureclient.event.EventBus;
import net.fabricmc.api.ModInitializer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * FutureClient - Modern Fabric PvP Client for Minecraft 1.20.1
 * Lunar Client-style experience with 1.8.9 PvP feel
 */
public class FutureClient implements ModInitializer {
    public static final String MOD_ID = "futureclient";
    public static final String MOD_NAME = "FutureClient";
    public static final String VERSION = "1.0.0";
    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_NAME);
    
    private static FutureClient instance;
    private ModuleManager moduleManager;
    private ConfigManager configManager;
    private GuiManager guiManager;
    private EventBus eventBus;
    
    @Override
    public void onInitialize() {
        instance = this;
        LOGGER.info("Initializing {} v{}", MOD_NAME, VERSION);
        
        // Initialize core systems
        eventBus = new EventBus();
        configManager = new ConfigManager();
        moduleManager = new ModuleManager();
        guiManager = new GuiManager();
        
        // Load configuration
        configManager.loadConfig();
        
        // Initialize modules
        moduleManager.initialize();
        
        LOGGER.info("{} initialized successfully!", MOD_NAME);
    }
    
    public static FutureClient getInstance() {
        return instance;
    }
    
    public ModuleManager getModuleManager() {
        return moduleManager;
    }
    
    public ConfigManager getConfigManager() {
        return configManager;
    }
    
    public GuiManager getGuiManager() {
        return guiManager;
    }
    
    public EventBus getEventBus() {
        return eventBus;
    }
}

File: src/main/kotlin/com/futureclient/FutureClient.kt
package com.futureclient

import com.futureclient.api.config.ConfigManager
import com.futureclient.api.module.ModuleManager
import com.futureclient.impl.gui.click.ClickGui
import com.futureclient.impl.modules.render.CustomMenu
import net.minecraft.client.Minecraft
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraftforge.fml.common.gameevent.InputEvent
import net.minecraftforge.fml.common.gameevent.TickEvent
import org.lwjgl.input.Keyboard
import java.io.File

@Mod(modid = FutureClient.MODID, version = FutureClient.VERSION, name = FutureClient.NAME)
class FutureClient {

    companion object {
        const val MODID = "futureclient"
        const val NAME = "FutureClient"
        const val VERSION = "1.0"

        @JvmStatic lateinit var instance: FutureClient
        val mc: Minecraft = Minecraft.getMinecraft()
    }

    lateinit var moduleManager: ModuleManager
    lateinit var configManager: ConfigManager
    lateinit var clickGui: ClickGui

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent) {
        instance = this
        val directory = File(mc.mcDataDir, "FutureClient")
        if (!directory.exists()) directory.mkdirs()

        moduleManager = ModuleManager()
        configManager = ConfigManager(File(directory, "config.json"))
        clickGui = ClickGui()

        MinecraftForge.EVENT_BUS.register(this)
        MinecraftForge.EVENT_BUS.register(moduleManager)
        
        // Initial load
        configManager.loadConfig()
    }

    @SubscribeEvent
    fun onKeyInput(event: InputEvent.KeyInputEvent) {
        if (Keyboard.getEventKeyState()) {
            val key = Keyboard.getEventKey()
            if (key == Keyboard.KEY_NONE) return
            
            // Handle Module Toggles
            moduleManager.getModules().filter { it.key == key }.forEach { it.toggle() }

            // Open ClickGUI (Default Right Shift)
            if (key == Keyboard.KEY_RSHIFT) {
                mc.displayGuiScreen(clickGui)
            }
        }
    }
    
    // Shut down hook to save config
    @SubscribeEvent
    fun onTick(event: TickEvent.ClientTickEvent) {
       // logic that runs every tick if needed globally
    }
}
